### ISSUE TITLE
Why not 1-sqrt for WSD decay?

### ISSUE THREAD

**Issue by slashedstar** [kozistr/pytorch_optimizer#382]
In the section 4 of the paper referenced for Warmup-Stable-Decay learning rate scheduler (https://arxiv.org/abs/2405.18392) they mention 1-sqrt being the best cooldown function, on page 20 they further demonstrate how it outperforms other functions so why the choice of cosine for WSD cooldown? https://github.com/kozistr/pytorch_optimizer/blob/main/pytorch_optimizer/lr_scheduler/wsd.py#L33

**Comment by kozistr** [comment]
@slashedstar hi. thanks for raising this. At that time, as far as I remembered, I first looked into the [MiniCPM experiment note](https://shengdinghu.notion.site/MiniCPM-Unveiling-the-Potential-of-End-side-Large-Language-Models-d4d3a8c426424654a4e80e42a711cb20) and the [PR in Transformer](https://github.com/huggingface/transformers/pull/30231), and I just added it here right away without checking the WSD paper honestly.

Of course, We could add `1-sqrt` and other cooldown functions too, and make `1-sqrt` cooldown function as a default one cause it's a best choice followed by the paper.

I'll work on it sometime later and thanks again for asking :)
